pyaedt/filtersolutions_core/ideal_response.py

```diff
@@ -0,0 +1,458 @@
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from ctypes import POINTER
+from ctypes import byref
+from ctypes import c_bool
+from ctypes import c_char_p
+from ctypes import c_double
+from ctypes import c_int
+from enum import Enum
+import math
+
+import pyaedt
+from pyaedt.filtersolutions_core.graph_setup import GraphSetup
+
+
+class FrequencyResponseColumn(Enum):
+    """Provides an enum of frequency response parameters.
+
+    Attributes:
+    MAGNITUDE_DB: Represents the frequency response magnitude in dB.
+    PHASE_DEG: Represents the frequency response phase in degree.
+    GROUP_DELAY: Represents the frequency response group delay.
+    PHASE_RAD: Represents the frequency response phase in radian.
+    MAGNITUDE_ARITH: Represents the frequency response magnitude.
+    MAGNITUDE_REAL: Represents the real part of frequency response  magnitude .
+    MAGNITUDE_IMAG: Represents the imaginary part of frequency response  magnitude.
+    PHASE_DEV_DEG: Represents the frequency response phase deviation in degree.
+    PHASE_DEV_RAD: Represents the frequency response phase deviation in radian.
+    FREQUENCY: Represents the frequency response frequency parameter.
+    """
+
+    MAGNITUDE_DB = 0
+    PHASE_DEG = 1
+    GROUP_DELAY = 2
+    PHASE_RAD = 4
+    MAGNITUDE_ARITH = 3
+    MAGNITUDE_REAL = 5
+    MAGNITUDE_IMAG = 6
+    PHASE_DEV_DEG = 7
+    PHASE_DEV_RAD = 8
+    FREQUENCY = 9
+
+
+class TimeResponseColumn(Enum):
+    """Provides an enum of time response parameters.
+
+    Attributes:
+    STEP_RESPONSE: Represents the step time response.
+    RAMP_RESPONSE: Represents the ramp time response.
+    IMPULSE_RESPONSE: Represents the impulse time response.
+    STEP_RESPONSE_DB: Represents the step time response in dB.
+    RAMP_RESPONSE_DB: Represents the ramp time response in dB.
+    IMPULSE_RESPONSE_DB: Represents the impulse time response in dB.
+    TIME: Represents time response the time parameter.
+    """
+
+    STEP_RESPONSE = 0
+    RAMP_RESPONSE = 1
+    IMPULSE_RESPONSE = 2
+    STEP_RESPONSE_DB = 3
+    RAMP_RESPONSE_DB = 4
+    IMPULSE_RESPONSE_DB = 5
+    TIME = 6
+
+
+class SParametersResponseColumn(Enum):
+    """Provides an enum of S parameters.
+
+    Attributes:
+    S21_DB: Represents the S21 parameter in dB.
+    S11_DB: Represents the S11 parameter in dB.
+    S21_ARITH: Represents the S21 parameter.
+    S11_ARITH: Represents the S11 parameter.
+    FREQUENCY: Represents the S parameters frequency parameter.
```

Comment:
```suggestion
    **Attributes:**

    - S21_DB: Represents the S21 parameter in dB.
    - S11_DB: Represents the S11 parameter in dB.
    - S21_ARITH: Represents the S21 parameter.
    - S11_ARITH: Represents the S11 parameter.
    - FREQUENCY: Represents the S parameters' frequency parameter.
```
I think you need either the possessive apostrophe here or to rewrite to "Represents the frequency parameter of the S parameters."